org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ProjectUtils.java

```diff
@@ -349,6 +349,7 @@ public static void updateBinaries(IJavaProject javaProject, Map<Path, IPath> lib
 		if (!Arrays.equals(rawClasspath, newClasspath)) {
 			javaProject.setRawClasspath(newClasspath, monitor);
 		}
+		javaProject.getProject().refreshLocal(IResource.DEPTH_ZERO, monitor);
```

Comment:
When the resource in jdt is out of sync with the local file system, InvisibleProjectBuildSupport#fileChanged also invoke resource.refreshLocal to sync the model, what's the purpose of this new line?

Reply:
This added line does not fully solved the problem, I'll update the PR later